src/session.rs

```diff
@@ -58,6 +58,12 @@ impl Session {
     /// `"amqp://localhost/"` and it will connect to rabbitmq server, running on `localhost` on port `65535`,
     /// with login `"guest"`, password: `"guest"` to vhost `"/"`
     pub fn open_url(url_string: &str) -> AMQPResult<Session> {
+        fn decode(string: &str) -> String {
+            let input: Vec<u8> = string.bytes().collect();
```

Comment:
You can do: `string.as_bytes()`, which will give you &str directly.
And also we might want to do lossy_utf8_percent_decode or return Option<String>,
So basically it will become:
```rust
let input = string.as_bytes();
percent_encoding::lossy_utf8_percent_decode(input)
```